docs/proposals/06-immutable-etcd-backups.md

```diff
@@ -0,0 +1,315 @@
+**Proposed Solution:**
+
+A new `EtcdOperatorTask` called `EtcdSnapshotImmutabilityExtension` will be created as defined in the operator tasks framework. This new `EtcdOperatorTask` extends the immutability period by deploying a job which uploads another copy of the latest snapshot to the object store.
+
+A full snapshot is taken before hibernating the ETCD cluster. This is to ensure that no state maintained in the etcd cluster is lost before hibernation.
+
+**Implementation Details:**
+
+- **Introduce the `extend-immutability` command to etcdbrctl**:
+  - etcd-backup-restore will be enhanced to support a new command `extend-immutability` which does the following:
+    - Downloads the latest full snapshot from the object store.
+    - Replaces the Unix epoch in the file-name of the downloaded snapshot to contain the time at which the file completes downloading.
+    - Uploads this newly renamed snapshot to the same object store.
+    - Renews the full snapshot lease after the upload is successful.
+
+    The immutability period of an object begins from the moment of upload, thus extending the immutability period of the latest snapshot. Renaming the snapshot is necessary since the downloaded snapshot can not simply be re-uploaded as uploading with the same name would be an attempt at modifying an already existing snapshot, which is disallowed.
+
+    This command could either be implemented standalone, or could be implemented as a wrapper over the `copy` command of `etcdbrctl` by extending the functionality of the `copy` command accordingly.
+- **Introduce the `garbage-collect` command to etcdbrctl**:
+  - etcd-backup-restore will be enhanced to support a new command `garbage-collect` which does the following:
+    - Perform garbage collection of the snapshots in the object store according to the policy specified with the `--garbage-collection-policy` flag.  
+
+    This functionality is needed since it would be necessary to garbage collect the (identical final) snapshots that are (re)uploaded in order to ensure that there is always a snapshot which is immutable.
+- **Update `Etcd` CRD:**
+  - Add `etcd.spec.hibernation`:  
+    Since there are situations outside of hibernation where the number of replicas of the statefulset would have to be scaled to zero, there needs to be an explicit way in which it is conveyed to etcd-druid that the etcd cluster is being hibernated. This can be achieved by extending the `Etcd` CRD by including a new field in the `spec` called `hibernated`.
+
+    ```yaml
+    hibernation:
+      enabled: <bool>
+    ```
+
+  - Add `etcd.status.hibernatedAt`:  
+    This field conveys information about whether the cluster has been successfully hibernated after a reconciliation, and the time at which it entered hibernation. This field is cleared when the cluster is woken up from hibernation.
+
+    ```yaml
+    hibernatedAt: <hibernation-time>
+    ```
+
+  - Add `immutableSettings.retentionType` under `etcd.spec.backup.store`.
```

Comment:
> Also, if we're keeping this field, I would rename it to immutabilitySettings. The term immutableSettings actually means making the settings itself immutable, rather than talking about the settings related to immutability.

I have renamed `immutabilitySettings ` as `immutability` to be consistent as requested in https://github.com/gardener/gardener/issues/10866#issuecomment-2481911583